### DISCUSSION TITLE
Zoom steps weird?

### DISCUSSION THREAD

**Discussion by PabloDevGit** [godotengine/godot-proposals#6082] [2D]
### Describe the project you are working on

Not important.

### Describe the problem or limitation you are having in your project

Simply increase the zoom in steps of 25% or 50%.
Currently if you increase the zoom from 100%, the zoom goes to 141.4%, why these strange values?.

![GIF](https://user-images.githubusercontent.com/19652075/211866815-3a72c5de-7268-471e-b25a-4a1c27d4dff0.gif)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The zoom moves in steps of 25% or 50%

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The zoom moves in steps of 25% or 50%

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

No

**Comment by Calinou** [comment]
The zoom step is `zoom * sqrt(2)` by design (or `zoom / sqrt(2)` when zooming out), so that clicking the button twice always doubles the zoom level.

**Reply by rylydou** [reply]
I see why the zoom system is the way it is right now but I think there should be an option to zoom in steps of 100% so that things stay pixel perfect in pixel art games.

Maybe it could be implemented with a context menu from right clicking the zoom percent?
Or maybe by right-clicking the plus and minus buttons can step by 100% while left-clicking still doubles?

**Reply by Calinou** [reply]
In 3.5 and later, hold <kbd>Alt</kbd> and use the mouse wheel, or use the <kbd>1</kbd>-<kbd>5</kbd> number keys to zoom to specific integer zoom factors (like in GIMP).
